tools/autograd/gen_python_functions.py

```diff
@@ -613,17 +614,43 @@ def group_declarations(declarations):
 # This function declares a partial order on declarations, and sorts them according
 # to its linear extension. This is necessary, because there's some ambiguity in the
 # choice of overload, and we want a different order.
+#
+# See Note[Order of overloads matters]
 def sort_declarations(grouped_decls):
+
+    # TODO: This is a hack!
+    #
+    # For some reason, when you specify a Scalar argument in a native
+    # function, you get a Declarations.yaml entry that looks like this:
+    #
+    #   - default: 1
+    #     dynamic_type: Scalar
+    #     is_nullable: false
+    #     kwarg_only: true
+    #     name: alpha
+    #     type: Scalar
+    #
+    # This is contrast to when there is a 'real' argument in TH
+    # Declarations.cwrap; this gets (correctly?) translated into
+    # dynamic_type: real, and type: Scalar.  I would like to fix this
+    # at the source but I have never understood what dynamic_type is
+    # supposed to be.
+    def normalized_dynamic_type(arg):
+        if arg['dynamic_type'] == 'real':
```

Comment:
why do you actually need this?  I don't really know what `dynamic_type` is trying to get across here, so it's not obvious to me that it's wrong.  Can you just change is_coord_smaller to do what you want?

Reply:
I can't only change `is_coord_smaller`, because there are equality tests done on `dynamic_type`, and if `real` and `Scalar` don't compare as equal I get the wrong sort. Maybe I should just fix this at the source and make sure everything works out OK. What do you think?

Reply:
not sure what you mean by at the source, do you mean so the native and non-native declarations entries are the same?  I find it hard to argue about this without really understanding what `dynamic_type` is supposed to represent.

Reply:
Yep, that's what I mean. Pinging @zdevito about the meaning of `dynamic_type` versus `type`.